app/(gcforms)/[locale]/(form administration)/form-builder/[id]/responses/[statusFilter]/components/ManageFormAccessDialog/ManageUsers.tsx

```diff
@@ -0,0 +1,262 @@
+  };
+
+  /**
+   * Add an error to the list
+   * @param error
+   */
+  const handleAddError = (error: string) => {
+    setErrors((prevErrors: string[]) => [...prevErrors, error]);
+  };
+
+  /**
+   * Handle adding one or more emails addresses to the list.
+   * Multiple emails can be delimited by comma or space.
+   * Emails are validated before being added to the list.
+   *
+   * @param emails
+   */
+  const handleAddEmail = (emails: string) => {
+    if (!emails) return;
+    setErrors([]);
+
+    const emailArray = emails.split(/[\s,]+/).map((email) => email.trim());
+    const validEmails = emailArray.filter((email: string) => isValidEmail(email));
+
+    setEmailList([...emailList, ...validEmails]);
+    setSelectedEmail("");
+  };
+
+  /**
+   * Remove an email from the list
+   * @param email
+   */
+  const handleRemoveEmail = (email: string) => {
+    setEmailList(emailList.filter((e) => e !== email));
+  };
+
+  const [loading, setLoading] = useState(true);
+
+  /**
+   * Fetch users with access to the form
+   */
+  const fetchUsersWithAccess = useCallback(async () => {
+    const users = await getTemplateUsers(formId);
+
+    // Make sure the logged-in user is at the top of the list
+    const sorted = [...users].sort((a, b) => {
+      if (a.email === loggedInUserEmail) return -1;
+      if (b.email === loggedInUserEmail) return 1;
+      return 0;
+    });
+
+    setUsersWithAccess(sorted || []);
+    setLoading(false);
+  }, [formId, loggedInUserEmail]);
+
+  useEffect(() => {
+    fetchUsersWithAccess();
+  }, [fetchUsersWithAccess, formId, setUsersWithAccess]);
+
+  const handleRemoveUser = async (userId: string) => {
+    const result = await removeUserFromForm(userId, formId);
+    if (result.success) {
+      setUsersWithAccess(usersWithAccess.filter((user) => user.id !== userId));
+      return true;
+    }
+    return false;
+  };
+
+  const handleResendInvitation = async (email: string) => {
+    handleAddEmail(email);
+  };
+
+  const handleCancelInvitation = async (id: string): Promise<boolean> => {
+    cancelInvitation(id);
+    setUsersWithAccess(usersWithAccess.filter((user) => user.id !== id));
+    return true;
+  };
+
+  return (
+    <>
+      <section>
+        <label htmlFor="email" className="font-bold">
+          {t("addPeopleToShareAccess")}
+        </label>
+        <p className="pb-3">{t("mustBeAGovernmentAddress")}</p>
+
+        {errors.length > 0 && (
+          <div className="my-2 flex flex-wrap gap-2">
+            {errors.map((error, index) => (
+              <div
+                className="rounded-md border border-red-700 bg-red-100 px-2 text-red-700"
+                key={`error-${index}`}
+              >
+                {error}
+              </div>
+            ))}
+          </div>
+        )}
+
+        <div className="flex flex-wrap gap-2 border border-black p-2">
+          {emailList.map((email) => {
+            return (
+              <div
+                key={email}
```

Comment:
another key thing here --- again likely not really a problem but might as well fix per the others.

Reply:
https://github.com/cds-snc/platform-forms-client/pull/4392/commits/ccc3094a10047bb05e5fc7371c6a1fe635cd11dc